File: 0013-Roman-to-Integer/0013.py
class Solution:
    def romanToInt(self, s: str) -> int:
        # last index
        idx = len(s) - 1

        # answer
        result = 0

        # iterate the string from right to left
        while idx >= 0:
            if s[idx] == 'I':
                # IV -> 4, IX -> 9
                if idx < len(s) - 1 and (s[idx + 1] == 'V' or s[idx + 1] == 'X'):
                    result -= 1
                else:
                    result += 1
            elif s[idx] == 'V':
                result += 5
            elif s[idx] == 'X':
                # XL -> 40, XC -> 90
                if idx < len(s) - 1 and (s[idx + 1] == 'L' or s[idx + 1] == 'C'):
                    result -= 10
                else:
                    result += 10
            elif s[idx] == 'L':
                result += 50
            elif s[idx] == 'C':
                # CD -> 500, CM -> 900
                if idx < len(s) - 1 and (s[idx + 1] == 'D' or s[idx + 1] == 'M'):
                    result -= 100
                else: 
                    result += 100
            elif s[idx] == 'D':
                result += 500
            elif s[idx] == 'M':
                result += 1000
            idx -= 1

        return result

File: 0013-Roman-to-Integer/0013.java
class Solution {
    public int romanToInt(String s) {
        // last index
        int idx = s.length() - 1;
        int answer = 0;

        for (int i = idx; i >= 0; i--) {
            if (s.charAt(i) == 'I') {
                if (i < idx && (s.charAt(i + 1) == 'V' || s.charAt(i + 1) == 'X'))
                    answer -= 1;
                else
                    answer += 1;
            } else if (s.charAt(i) == 'V') {
                answer += 5;
            } else if (s.charAt(i) == 'X') {
                if (i < idx && (s.charAt(i + 1) == 'L' || s.charAt(i + 1) == 'C'))
                    answer -= 10;
                else
                    answer += 10;
            } else if (s.charAt(i) == 'L') {
                answer += 50;
            } else if (s.charAt(i) == 'C') {
                if (i < idx && (s.charAt(i + 1) == 'D' || s.charAt(i + 1) == 'M'))
                    answer -= 100;
                else
                    answer += 100;
            } else if (s.charAt(i) == 'D') {
                answer += 500;
            } else if (s.charAt(i) == 'M') {
                answer += 1000;
            }
        }
        return answer;
    }
}